### PR TITLE
feat(server): add colors module

### PR DESCRIPTION
**Pull request by dayongkr**
## Summary

- Add `es-toolkit/color` subpath for terminal color utilities
- Fixes known picocolors bugs: `FORCE_COLOR=0` handling, `process` ReferenceError, CI pipe detection
- Supports 16 basic colors, 256-color, RGB, and Hex
- Includes `stripAnsi`, `createColors`, `colorLevel`, `isColorSupported`
- Pre-configured `color` object for direct usage (`color.red('hello')`)
- FORCE_COLOR / NO_COLOR spec compliant
- 55 tests, 100% coverage
- Documentation in 4 languages (en, ko, ja, zh_hans)

## Test plan

- [ ] `yarn vitest run src/color/` — all 55 tests pass
- [ ] `yarn lint` — no errors in `src/color/`
- [ ] `tsc --noEmit` — no type errors
- [ ] `yarn build` — dist/color/ generated correctly
- [ ] Verify docs render at `/reference/color/color`

### CODE REVIEW

File: src/color/formatter.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * reset the background at line boundaries.
+ */
+export function createFormatter(open: string, close: string, isBackground: boolean): ColorFunction {
+  return (text: unknown): string => {
+    const str = toString(text);
+
+    if (str === '') {
+      return '';
+    }
+
```
Comment (raon0211) [inline src/color/formatter.ts]: What about requiring strings in `text` and removing this logic?

File: src/color/formatter.spec.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
+// @vitest-environment node
```
Comment (raon0211) [inline src/color/formatter.spec.ts]: There might be some edge cases here -- I think we need to make the test cases more complete.

File: src/color/createColors.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+ * @example
+ * import { createColors } from 'es-toolkit/color';
+ *
+ * const c = createColors(true);
+ * c.red('hello'); // '\x1b[31mhello\x1b[39m'
+ *
+ * const noColor = createColors(false);
+ * noColor.red('hello'); // 'hello'
+ */
+export function createColors(enabled?: boolean): Colors {
```
Comment (raon0211) [inline src/color/createColors.ts]: What about we make these into functions like this?

```
export function reset(str: string) {
  if (detectColorLevel()) {
    return str;
  }

  return formatWithColor(str, '\x1b[0m', '\x1b[0m');
}
```

Comment (raon0211) [reply src/color/createColors.ts]:
```typescript
import colors from 'es-toolkit/colors';
```

I think we also have to provide this too

Comment (raon0211) [reply src/color/createColors.ts]:
```
import * as colors from './colors';

export default colors;
```

File: src/color/createFormatter.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * Background colors get special newline handling because terminals
+ * reset the background at line boundaries.
+ */
+export function createFormatter(open: string, close: string, isBackground: boolean): ColorFunction {
+  return (text: string): string => {
+    let result = text;
+
+    // Re-open the outer style when a nested call with the same color
+    // inserts a close code inside the text.
+    if (open !== close && result.includes(close)) {
```
Comment (raon0211) [inline src/color/createFormatter.ts]: Why is `open !== close` needed here?

File: src/color/colorLevel.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+ * 1. FORCE_COLOR — User explicitly forces a specific color level (spec: force-color.org)
+ * 2. NO_COLOR — User explicitly disables all colors (spec: no-color.org)
+ * 3. TTY check — Non-interactive output (e.g. piped to file) gets no colors
+ * 4. Windows — Detects Windows Terminal (truecolor) vs legacy cmd.exe (basic)
+ * 5. COLORTERM — Set by terminals that support truecolor (e.g. iTerm2, Alacritty)
+ * 6. TERM — Set by terminals; "256color" suffix indicates 256-color support
+ * 7. Node.js hasColors() — Runtime API (Node 11.13+) for accurate detection
+ * 8. Default — Basic 16 colors if none of the above matched
+ */
+export function detectColorLevel(): ColorLevel {
```
Comment (raon0211) [inline src/color/colorLevel.ts]: I think we have to simplify this function.
